File: src/main/java/org/hwabeag/playerfly/inventorys/FlySettingGUI.java
package org.hwabeag.playerfly.inventorys;

import java.util.ArrayList;
import java.util.Objects;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.SkullMeta;
import org.hwabeag.playerfly.config.ConfigManager;

public class FlySettingGUI implements Listener {
    private final Inventory inv = Bukkit.createInventory((InventoryHolder)null, 9, "플라이");
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");

    private ItemStack getHead(String name) {
        ItemStack item = new ItemStack(Material.PLAYER_HEAD, 1, (short) 3);
        SkullMeta skull = (SkullMeta) item.getItemMeta();
        if (Objects.equals(PlayerFlyConfig.getString("플라이." + name + ".관리"), "온")) {
            skull.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a&l" + name + " &7님의 플라이 &a사용중"));
            skull.setOwner(name);
        } else if (Objects.equals(PlayerFlyConfig.getString("플라이." + name + ".관리"), "오프")) {
            skull.setDisplayName(ChatColor.translateAlternateColorCodes('&', "&a&l" + name + " &7님의 플라이 &c중지중"));
            skull.setOwner(name);
        }

        String TimeNumber = PlayerFlyConfig.getString("플라이." + name + ".남은시간");
        ArrayList<String> loreList = new ArrayList();
        if (PlayerFlyConfig.getBoolean("playerfly.fly-infinity")) {
            loreList.add(ChatColor.translateAlternateColorCodes('&', "&7&l- &a&l무제한 &7사용가능"));
        } else {
            loreList.add(ChatColor.translateAlternateColorCodes('&', "&7&l- &a&l" + TimeNumber + " &7분 동안 사용가능"));
        }
        loreList.add(ChatColor.translateAlternateColorCodes('&', "&7&l- &a&l클릭시 플라이를 활성화 혹 비활성화 할 수 있습니다."));
        skull.setLore(loreList);
        item.setItemMeta(skull);
        return item;
    }

    private void initItemSetting(Player p) {
        String name = p.getName();
        inv.setItem(4, getHead(name));
    }

    public FlySettingGUI(Player p) {
        initItemSetting(p);
    }

    public void open(Player player) {
        player.openInventory(inv);
    }
}

File: src/main/kotlin/org/hwabeag/playerfly/inventory/FlySettingGUI.kt
package org.hwabeag.playerfly.inventory

import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.SkullMeta
import org.hwabeag.playerfly.PlayerFlyPlugin

class FlySettingGUI(private val plugin: PlayerFlyPlugin) {
    companion object {
        const val TITLE = "플라이 설정"
    }

    fun open(player: Player) {
        val inv: Inventory = Bukkit.createInventory(null, 9, TITLE)
        inv.setItem(4, createHead(player))
        player.openInventory(inv)
    }

    private fun createHead(player: Player): ItemStack {
        val item = ItemStack(Material.PLAYER_HEAD)
        val meta = item.itemMeta as? SkullMeta ?: return item
        val data = plugin.databaseManager.getData(player.uniqueId, player.name)

        meta.owningPlayer = player
        meta.setDisplayName(
            if (data.enabled) {
                plugin.color("&a&l${player.name} &7님의 플라이: &a사용 중")
            } else {
                plugin.color("&a&l${player.name} &7님의 플라이: &c중지 중")
            }
        )

        val firstLine = if (plugin.config.getBoolean("playerfly.fly-infinity")) {
            plugin.color("&7&l- &a&l무제한 사용 가능")
        } else {
            plugin.color("&7&l- &a&l${data.minutes} &7분 사용 가능")
        }

        meta.lore = listOf(
            firstLine,
            plugin.color("&7&l- &a&l클릭 시 플라이를 켜거나 끌 수 있습니다.")
        )
        item.itemMeta = meta
        return item
    }
}
